main.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"context"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/pkg/errors"
+
+	"github.com/HarlamovBuldog/social-tournament-service/pkg/server"
+	"github.com/HarlamovBuldog/social-tournament-service/pkg/storage"
+	"gopkg.in/yaml.v3"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type config struct {
+	ConnStr    string `yaml:"conn_str"`
+	ServerPort string `yaml:"server_port"`
```

Comment:
why server port is str? maybe int?

Reply:
i've validate method where i check if server port is uint. I can do it here and field type uint, but then i will get not very informative err message from decode then.

Reply:
I would better use uint and use error.wrap to describe an error